configure.py

```diff
@@ -1663,7 +1663,14 @@ def get_clang_inline_threshold():
 globals().update(vars(args))
 
 total_memory = os.sysconf('SC_PAGE_SIZE') * os.sysconf('SC_PHYS_PAGES')
+# assuming each link job takes around 7GiB of memory without LTO
 link_pool_depth = max(int(total_memory / 7e9), 1)
+if args.lto:
+    # ThinLTO provides its own parallel linking, use 16GiB for RAM size used
+    # by each link job
+    depth_with_lto = max(int(total_memory / 16e9), 2)
```

Comment:
why is the second arg of max 2 here, while for the regular link pool it is 1

Reply:
1 because we need to make some progress.

2 because the regular link jobs which links tests does not use LTO, i wanted maximize the parallelism of the build process, so if we have a building host with, for instance 30GiB RAM, it should be able to run a LTO link job and a non-LTO link job, instead of a single LTO link job.

currently, we only link the scylla main executable in release build with LTO enabled. and because we build FatLTO objects for the release build, the link job also compiles the LLVM bit code at the link time, so linking the main executable could take much longer, so 2 jobs allows us to get the best out of the build prorcess.

Reply:
As the code is currently, on a host with 8GiB of RAM, link-depth == 1 without LTO and == 2 with LTO, which we expect will use more memory. This makes no sense to me.
Anyway, all our build machines have more than 8GiB of memory, so this is probably a non-issue but the inconsistency of the logic still irks me.